network/handlers/drain_test.go

```diff
@@ -155,8 +163,9 @@ func TestDrainMechanics(t *testing.T) {
 			drainer.ServeHTTP(w, req)
 		}
 	}
-	if mt.resetCalls != rc+3 {
-		t.Errorf("ResetCalls = %d, want: %d", mt.resetCalls, rc+1)
+	// Two more drains should have been called.
+	if got, want := mt.resetCalls, rc+2; got != want {
+		t.Errorf("ResetCalls = %d, want: %d", got, want)
```

Comment:
Move this into the `if` in the loop and do it in the same style?

Reply:
sure.